diff --git a/src/runtime/virtcontainers/qemu.go b/src/runtime/virtcontainers/qemu.go
index 51a46d272..98e52ec6a 100644
--- a/src/runtime/virtcontainers/qemu.go
+++ b/src/runtime/virtcontainers/qemu.go
@@ -2686,6 +2686,8 @@ func genericAppendPCIeRootPort(devices []govmmQemu.Device, number uint32, machin
 	default:
 		return devices
 	}
+	// Set both memSize32bit to 32Mb
+	memSize32bit = 32 * 1024 * 1024
 
 	for i := uint32(0); i < number; i++ {
 		devices = append(devices,
