diff --git a/src/runtime/virtcontainers/qemu.go b/src/runtime/virtcontainers/qemu.go
index b21ef94bb..d268d68f8 100644
--- a/src/runtime/virtcontainers/qemu.go
+++ b/src/runtime/virtcontainers/qemu.go
@@ -3449,6 +3449,8 @@ func genericAppendPCIeRootPort(devices []govmmQemu.Device, number uint32, machin
 				Chassis:       chassis,
 				Slot:          strconv.FormatUint(uint64(i), 10),
 				Multifunction: multiFunction,
+				MemReserve:    "256m",
+				Pref64Reserve: "512m",
 			},
 		)
 	}
@@ -3496,6 +3498,8 @@ func genericAppendPCIeSwitchPort(devices []govmmQemu.Device, number uint32, mach
 		Chassis:       "1",
 		Slot:          strconv.FormatUint(uint64(0), 10),
 		Multifunction: false,
+		MemReserve:    "256m",
+		Pref64Reserve: "512m",
 	}
 
 	devices = append(devices, pcieRootPort)
